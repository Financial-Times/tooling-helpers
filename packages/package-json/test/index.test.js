const loadPackageJson = require("../src/index.js");

let packageJson;

beforeEach(() => {
	packageJson = loadPackageJson({
		filepath: `${__dirname}/fixtures/test-package.json`
	});
});

describe("loadPackageJson", () => {
	test("throws an error if the `filepath` option is missing", () => {
		expect(() => {
			loadPackageJson();
		}).toThrowErrorMatchingSnapshot();
	});

	test("throws an error if is unable to read the specified file", () => {
		expect(() => {
			loadPackageJson({
				filepath: "/app/non-existent-package.json"
			});
		}).toThrowErrorMatchingSnapshot();
	});

	test("throws an error if is unable to parse the specified file as JSON", () => {
		/**
		 * We can't use `toThrowErrorMatchingSnapshot` here as the filepath in
		 * the error message is absolute, so it will be different in every
		 * environment that these tests are run on.
		 */
		expect(() => {
			loadPackageJson({
				filepath: `${__dirname}/fixtures/bad-package.json`
			});
		}).toThrowError();
	});

	test("returns object with methods for manipulating package.json", () => {
		const packageJson = loadPackageJson({
			filepath: `${__dirname}/fixtures/test-package.json`
		});
		expect(packageJson).toEqual(expect.any(Object));
		Object.values(packageJson).forEach(method => {
			expect(method).toEqual(expect.any(Function));
		});
	});

	test("loads the specified package.json file", () => {
		const packageJson = loadPackageJson({
			filepath: `${__dirname}/fixtures/test-package.json`
		});
		expect(packageJson.get().name).toEqual("ebi");
		expect(packageJson.get()).toMatchSnapshot();
	});
});

describe("get", () => {
	test("returns an object that represents the working state of the `package.json` object", () => {
		expect(packageJson.get()).toMatchSnapshot();

		packageJson.setField("name", "ebi v2");
		expect(packageJson.get().name).toEqual("ebi v2");
	});
});

describe("hasChangesToWrite", () => {
	test("returns true when there are pending changes to write to package.json", () => {
		packageJson.removeDependency({
			pkg: "prettier",
			field: "devDependencies"
		});
		expect(packageJson.hasChangesToWrite()).toEqual(true);
	});

	test("returns false when there are no pending changes to write to package.json", () => {
		expect(packageJson.hasChangesToWrite()).toEqual(false);
	});
});

describe("getField", () => {
	test("returns field contents if field exists", () => {
		expect(packageJson.getField("name")).toEqual("ebi");
	});

	test("returns undefined if field does not exist", () => {
		expect(packageJson.getField("private")).toEqual(undefined);
	});
});

describe("setField", () => {
	test("logs a field to write if field exists", () => {
		const changelogEntry = packageJson.setField("name", "ebi v2");
		expect(changelogEntry).toMatchSnapshot();
		expect(packageJson.get()).toMatchSnapshot();
	});

	test("logs a field to create if field does not exist", () => {
		const changelogEntry = packageJson.setField("private", true);
		expect(changelogEntry).toMatchSnapshot();
		expect(changelogEntry.previousValue).toEqual(undefined);
		expect(packageJson.get()).toMatchSnapshot();
	});
});

describe("requireDependency", () => {
	test("throws error if dependencyField does not exist", () => {
		expect(() => {
			packageJson.requireDependency({
				pkg: "jest",
				version: "24.0.0",
				field: "testDependencies"
			});
		}).toThrowErrorMatchingSnapshot();
	});

	test("updates version of existing dependency", () => {
		const changelogEntry = packageJson.requireDependency({
			pkg: "prettier",
			version: "1.16.4",
			field: "devDependencies"
		});
		expect(changelogEntry).toMatchSnapshot();
		expect(changelogEntry.previousValue).toEqual("1.16.3");
		expect(packageJson.get()).toMatchSnapshot();
	});

	test("creates new dependency if absent", () => {
		const changelogEntry = packageJson.requireDependency({
			pkg: "nock",
			version: "10.0.6",
			field: "devDependencies"
		});
		expect(changelogEntry).toMatchSnapshot();
		expect(changelogEntry.alreadyExisted).toEqual(false);
		expect(packageJson.get()).toMatchSnapshot();
	});

	test("creates field and new dependency if both are absent", () => {
		const changelogEntry = packageJson.requireDependency({
			pkg: "nock",
			version: "10.0.6",
			field: "peerDependencies"
		});
		expect(changelogEntry).toMatchSnapshot();
		expect(changelogEntry.alreadyExisted).toEqual(false);
		expect(packageJson.get()).toMatchSnapshot();
	});
});

describe("removeDependency", () => {
	test("throws error if `field` is not a valid field for dependencies", () => {
		expect(() => {
			packageJson.removeDependency({
				pkg: "ebi",
				version: "1.1.0",
				field: "testDependencies"
			});
		}).toThrowErrorMatchingSnapshot();
	});

	test("returns boolean false if pkg does not exist", () => {
		expect(
			packageJson.removeDependency({
				pkg: "ebi",
				version: "1.1.0",
				field: "devDependencies"
			})
		).toEqual(false);
	});

	test("returns boolean false if `field` is valid but does not exist", () => {
		expect(
			packageJson.removeDependency({
				pkg: "ebi",
				version: "1.1.0",
				field: "peerDependencies"
			})
		).toEqual(false);
	});

	test("removes existing pkg", () => {
		const changelogEntry = packageJson.removeDependency({
			pkg: "prettier",
			field: "devDependencies"
		});
		expect(changelogEntry).toMatchSnapshot();
		expect(changelogEntry.previousValue).toEqual("1.16.3");
		expect(changelogEntry.meta.pkg).toEqual("prettier");

		expect(packageJson.get()).toMatchSnapshot();
	});
});

describe("requireScript", () => {
	test("creates new script if absent", () => {
		const changelogEntry = packageJson.requireScript({
			stage: "unit-test",
			command: "test"
		});
		expect(changelogEntry).toMatchSnapshot();
		expect(changelogEntry.alreadyExisted).toEqual(false);
		expect(changelogEntry.meta.stage).toEqual("unit-test");
		expect(packageJson.get()).toMatchSnapshot();
	});

	test("creates new script if no previous scripts", () => {
		const packageJsonWithoutScripts = loadPackageJson({
			filepath: `${__dirname}/fixtures/test-no-scripts-package.json`
		});
		const changelogEntry = packageJsonWithoutScripts.requireScript({
			stage: "test",
			command: "npm run unit-test"
		});
		expect(changelogEntry).toMatchSnapshot();
		expect(changelogEntry.alreadyExisted).toEqual(false);
		const workingContents = packageJsonWithoutScripts.get();
		expect(workingContents).toMatchSnapshot();
		expect(workingContents.scripts.test).toEqual("npm run unit-test");
	});

	test("replaces script if present", () => {
		const changelogEntry = packageJson.requireScript({
			stage: "test",
			command: "npm run unit-test"
		});
		expect(changelogEntry).toMatchSnapshot();
		expect(changelogEntry.alreadyExisted).toEqual(true);
		expect(changelogEntry.meta.stage).toEqual("test");
		expect(packageJson.get()).toMatchSnapshot();
	});
});
