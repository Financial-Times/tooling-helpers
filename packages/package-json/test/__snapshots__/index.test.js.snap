// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadPackageJson loads the specified package.json file 1`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
    "prettier": "1.16.3",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi",
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "eslint-check": "eslint --print-config . | eslint-config-prettier-check",
    "lint": "eslint *.js src/ lib/ test/",
    "lint-fix": "eslint --fix *.js src/ lib/ test/",
    "test": "npm run lint && npm run unit-test",
  },
  "version": "0.0.0",
}
`;

exports[`loadPackageJson throws an error if is unable to parse the specified file as JSON 1`] = `
"package-json#loadPackageJson: Unable to read and parse file: /home/simonplend/dev/work/clients/financial-times/tooling-helpers/packages/package-json/test/fixtures/bad-package.json

Unexpected token e in JSON at position 14"
`;

exports[`loadPackageJson throws an error if is unable to read the specified file 1`] = `
"package-json#loadPackageJson: Unable to read and parse file: /home/simonplend/dev/work/clients/financial-times/tooling-helpers/packages/package-json/test/fixtures/non-existent-package.json

ENOENT: no such file or directory, open '/home/simonplend/dev/work/clients/financial-times/tooling-helpers/packages/package-json/test/fixtures/non-existent-package.json'"
`;

exports[`loadPackageJson throws an error if the \`filepath\` option is missing 1`] = `"package-json#constructor: \`filepath\` option must be specified"`;

exports[`removeDependency removes existing pkg 1`] = `
Object {
  "alreadyExisted": false,
  "event": "removeDependency",
  "field": "devDependencies",
  "meta": Object {
    "pkg": "prettier",
  },
  "previousValue": "1.16.3",
}
`;

exports[`removeDependency removes existing pkg 2`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi",
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "eslint-check": "eslint --print-config . | eslint-config-prettier-check",
    "lint": "eslint *.js src/ lib/ test/",
    "lint-fix": "eslint --fix *.js src/ lib/ test/",
    "test": "npm run lint && npm run unit-test",
  },
  "version": "0.0.0",
}
`;

exports[`removeDependency throws error if \`field\` is not a valid field for dependencies 1`] = `"package-json#removeDependency: Invalid field specified 'testDependencies'. Valid fields: dependencies, devDependencies, optionalDependencies, peerDependencies"`;

exports[`requireDependency creates field and new dependency if both are absent 1`] = `
Object {
  "alreadyExisted": false,
  "event": "requireDependency",
  "field": "peerDependencies",
  "meta": Object {
    "pkg": "nock",
    "version": "10.0.6",
  },
}
`;

exports[`requireDependency creates field and new dependency if both are absent 2`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
    "prettier": "1.16.3",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi",
  "peerDependencies": Object {
    "nock": "10.0.6",
  },
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "eslint-check": "eslint --print-config . | eslint-config-prettier-check",
    "lint": "eslint *.js src/ lib/ test/",
    "lint-fix": "eslint --fix *.js src/ lib/ test/",
    "test": "npm run lint && npm run unit-test",
  },
  "version": "0.0.0",
}
`;

exports[`requireDependency creates new dependency if absent 1`] = `
Object {
  "alreadyExisted": false,
  "event": "requireDependency",
  "field": "devDependencies",
  "meta": Object {
    "pkg": "nock",
    "version": "10.0.6",
  },
}
`;

exports[`requireDependency creates new dependency if absent 2`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
    "nock": "10.0.6",
    "prettier": "1.16.3",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi",
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "eslint-check": "eslint --print-config . | eslint-config-prettier-check",
    "lint": "eslint *.js src/ lib/ test/",
    "lint-fix": "eslint --fix *.js src/ lib/ test/",
    "test": "npm run lint && npm run unit-test",
  },
  "version": "0.0.0",
}
`;

exports[`requireDependency throws error if dependencyField does not exist 1`] = `"package-json#requireDependency: Invalid field specified 'testDependencies'. Valid fields: dependencies, devDependencies, optionalDependencies, peerDependencies"`;

exports[`requireDependency updates version of existing dependency 1`] = `
Object {
  "alreadyExisted": true,
  "event": "requireDependency",
  "field": "devDependencies",
  "meta": Object {
    "pkg": "prettier",
    "version": "1.16.4",
  },
  "previousValue": "1.16.3",
}
`;

exports[`requireDependency updates version of existing dependency 2`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
    "prettier": "1.16.4",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi",
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "eslint-check": "eslint --print-config . | eslint-config-prettier-check",
    "lint": "eslint *.js src/ lib/ test/",
    "lint-fix": "eslint --fix *.js src/ lib/ test/",
    "test": "npm run lint && npm run unit-test",
  },
  "version": "0.0.0",
}
`;

exports[`requireScript creates new script if absent 1`] = `
Object {
  "alreadyExisted": false,
  "event": "requireScript",
  "field": "scripts",
  "meta": Object {
    "lifecycleEvent": "unit-test",
  },
}
`;

exports[`requireScript creates new script if absent 2`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
    "prettier": "1.16.3",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi",
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "eslint-check": "eslint --print-config . | eslint-config-prettier-check",
    "lint": "eslint *.js src/ lib/ test/",
    "lint-fix": "eslint --fix *.js src/ lib/ test/",
    "test": "npm run lint && npm run unit-test",
    "unit-test": "test",
  },
  "version": "0.0.0",
}
`;

exports[`requireScript creates new script if no previous scripts 1`] = `
Object {
  "alreadyExisted": false,
  "event": "requireScript",
  "field": "scripts",
  "meta": Object {
    "lifecycleEvent": "test",
  },
}
`;

exports[`requireScript creates new script if no previous scripts 2`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
    "prettier": "1.16.3",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi",
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "test": "npm run unit-test",
  },
  "version": "0.0.0",
}
`;

exports[`requireScript replaces script if present 1`] = `
Object {
  "alreadyExisted": true,
  "event": "requireScript",
  "field": "scripts",
  "meta": Object {
    "lifecycleEvent": "test",
  },
}
`;

exports[`requireScript replaces script if present 2`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
    "prettier": "1.16.3",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi",
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "eslint-check": "eslint --print-config . | eslint-config-prettier-check",
    "lint": "eslint *.js src/ lib/ test/",
    "lint-fix": "eslint --fix *.js src/ lib/ test/",
    "test": "npm run unit-test",
  },
  "version": "0.0.0",
}
`;

exports[`setField logs a field to create if field does not exist 1`] = `
Object {
  "alreadyExisted": false,
  "event": "setField",
  "field": "private",
  "meta": Object {},
}
`;

exports[`setField logs a field to create if field does not exist 2`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
    "prettier": "1.16.3",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi",
  "private": true,
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "eslint-check": "eslint --print-config . | eslint-config-prettier-check",
    "lint": "eslint *.js src/ lib/ test/",
    "lint-fix": "eslint --fix *.js src/ lib/ test/",
    "test": "npm run lint && npm run unit-test",
  },
  "version": "0.0.0",
}
`;

exports[`setField logs a field to write if field exists 1`] = `
Object {
  "alreadyExisted": false,
  "event": "setField",
  "field": "name",
  "meta": Object {},
  "previousValue": "ebi",
}
`;

exports[`setField logs a field to write if field exists 2`] = `
Object {
  "bin": "./bin/ebi.js",
  "bugs": Object {
    "url": "https://github.com/Financial-Times/ebi/issues",
  },
  "contributors": Array [
    Object {
      "email": "jennifer.johnson@ft.com",
      "name": "Jennifer Johnson",
    },
    Object {
      "email": "jennifer.shepherd@ft.com",
      "name": "Jennifer Shepherd",
    },
    Object {
      "email": "tak.tran@ft.com",
      "name": "Tak Tran",
    },
  ],
  "dependencies": Object {
    "@octokit/plugin-throttling": "^2.4.0",
    "@octokit/rest": "^16.15.0",
    "lodash": "^4.17.11",
    "nock": "^10.0.6",
    "yargs": "^13.1.0",
  },
  "description": "A command line tool that searches files within GitHub repositories",
  "devDependencies": Object {
    "eslint": "^5.14.1",
    "eslint-config-prettier": "^4.0.0",
    "eslint-plugin-no-only-tests": "^2.1.0",
    "husky": "^1.3.1",
    "jest": "^24.1.0",
    "prettier": "1.16.3",
  },
  "engines": Object {
    "node": "10.15.0",
  },
  "homepage": "https://github.com/Financial-Times/ebi#readme",
  "husky": Object {
    "hooks": Object {
      "pre-commit": "npm run lint",
    },
  },
  "license": "MIT",
  "name": "ebi v2",
  "repository": Object {
    "type": "git",
    "url": "git+https://github.com/Financial-Times/ebi.git",
  },
  "scripts": Object {
    "eslint-check": "eslint --print-config . | eslint-config-prettier-check",
    "lint": "eslint *.js src/ lib/ test/",
    "lint-fix": "eslint --fix *.js src/ lib/ test/",
    "test": "npm run lint && npm run unit-test",
  },
  "version": "0.0.0",
}
`;
